UserStore.cs

```diff
@@ -7,543 +7,492 @@
+
+		/// <summary>
+		///     Adds the claim asynchronous.
+		/// </summary>
+		/// <param name="user">The user.</param>
+		/// <param name="claim">The claim.</param>
+		/// <returns>Task.</returns>
+		/// <exception cref="System.ArgumentNullException">user</exception>
+		public Task AddClaimAsync(TUser user, Claim claim) {
+			ThrowIfDisposed();
+			if (user == null)
+				throw new ArgumentNullException("user");
+
+			if (!user.Claims.Any(x => x.ClaimType == claim.Type && x.ClaimValue == claim.Value)) {
+				user.Claims.Add(new IdentityUserClaim {
+					ClaimType = claim.Type,
+					ClaimValue = claim.Value
+				});
+			}
+
+
+			return Task.FromResult(0);
+		}
+
+		/// <summary>
+		///     Gets the claims asynchronous.
+		/// </summary>
+		/// <param name="user">The user.</param>
+		/// <returns>Task{IList{Claim}}.</returns>
+		/// <exception cref="System.ArgumentNullException">user</exception>
+		public Task<IList<Claim>> GetClaimsAsync(TUser user) {
+			ThrowIfDisposed();
+			if (user == null)
+				throw new ArgumentNullException("user");
+
+			IList<Claim> result = user.Claims.Select(c => new Claim(c.ClaimType, c.ClaimValue)).ToList();
+			return Task.FromResult(result);
+		}
+
+		/// <summary>
+		///     Removes the claim asynchronous.
+		/// </summary>
+		/// <param name="user">The user.</param>
+		/// <param name="claim">The claim.</param>
+		/// <returns>Task.</returns>
+		/// <exception cref="System.ArgumentNullException">user</exception>
+		public Task RemoveClaimAsync(TUser user, Claim claim) {
+			ThrowIfDisposed();
+			if (user == null)
+				throw new ArgumentNullException("user");
+
+			user.Claims.RemoveAll(x => x.ClaimType == claim.Type && x.ClaimValue == claim.Value);
+			return Task.FromResult(0);
+		}
+
+
+		/// <summary>
+		///     Creates the user asynchronous.
+		/// </summary>
+		/// <param name="user">The user.</param>
+		/// <returns>Task.</returns>
+		/// <exception cref="System.ArgumentNullException">user</exception>
+		public Task CreateAsync(TUser user) {
+			ThrowIfDisposed();
+			if (user == null)
+				throw new ArgumentNullException("user");
+
+			db.GetCollection<TUser>(collectionName).InsertOneAsync(user);
+
+			return Task.FromResult(user);
+		}
+
+		/// <summary>
+		///     Deletes the user asynchronous.
+		/// </summary>
+		/// <param name="user">The user.</param>
+		/// <returns>Task.</returns>
+		/// <exception cref="System.ArgumentNullException">user</exception>
+		public Task DeleteAsync(TUser user) {
+			ThrowIfDisposed();
+			if (user == null)
+				throw new ArgumentNullException("user");
+
+			db.GetCollection<TUser>(collectionName).DeleteOneAsync(x => x.Id == user.Id);
+			return Task.FromResult(true);
+		}
+
+		/// <summary>
+		///     Finds the by identifier asynchronous.
+		/// </summary>
+		/// <param name="userId">The user identifier.</param>
+		/// <returns>Task{`0}.</returns>
+		public Task<TUser> FindByIdAsync(string userId) {
+			ThrowIfDisposed();
+
+			return Task.FromResult(
```

Comment:
The return statement can be simplified to:
`return db.GetCollection<TUser>(collectionName).Find(u => u.Id == userId).SingleOrDefaultAsync();`